src/hooks/index.ts

```diff
@@ -92,42 +85,44 @@ export function useEagerConnect() {
     }
   }, [activate, setWallet])
 
-  useEffect(() => {
-    const eagerConnect = async () => {
-      if (!triedSafe && connector === gnosisSafe) {
-        gnosisSafe.isSafeApp().then(loadedInSafe => {
-          if (loadedInSafe) {
-            activate(gnosisSafe, undefined, true).catch(() => {
-              setTriedSafe(true)
-            })
-          } else {
+  const eagerConnect = useCallback(async () => {
+    if (!triedSafe && connector === gnosisSafe) {
+      gnosisSafe.isSafeApp().then(loadedInSafe => {
+        if (loadedInSafe) {
+          activate(gnosisSafe, undefined, true).catch(() => {
             setTriedSafe(true)
-          }
-        })
-      } else if (connector?.isAuthorized) {
-        connector.isAuthorized?.().then(isAuthorized => {
-          if (isAuthorized) {
-            activate(connector, undefined, true).catch(() => {
-              setWallet(null)
-              setTried(true)
-            })
-          } else {
+          })
+        } else {
+          setTriedSafe(true)
+        }
+      })
+    } else if (connector?.isAuthorized) {
+      connector.isAuthorized?.().then(isAuthorized => {
+        if (isAuthorized) {
+          activate(connector, undefined, true).catch(() => {
             setWallet(null)
             setTried(true)
-          }
-        })
-      } else {
-        setWallet(null)
-        setTried(true)
-      }
+          })
+        } else {
+          setWallet(null)
+          setTried(true)
+        }
+      })
+    } else {
+      setWallet(null)
+      setTried(true)
     }
+  }, [activate, triedSafe, setTriedSafe, connector, setTried, setWallet])
+
+  useEffect(() => {
+    if (tried) return
```

Comment:
what benefit we get by moving `eagerConnect` from useEffect to outside as useCallback ?

Reply:
I moved it to usecallback to avoid some useeffect updates, `eagerConnect` uses some variables that were always updating and look more like activeMobile, but I don't know if this is necessary because there is that if in the useeffect if(tried) return